tools/Vis_PV.py

```diff
@@ -0,0 +1,286 @@
+
+# Set up the point of view (location, angle, etc.)
+
+def set_defaultCamera(renderView1):
+    '''
+    Set view and camera properties to a default 
+    specified below
+    '''
+    renderView1.ViewSize = [904, 501]
+    renderView1.AxesGrid = 'GridAxes3DActor'
+    renderView1.CenterOfRotation = [0.0, 0.0, -22.98529815673828]
+    renderView1.StereoType = 0
+    renderView1.CameraPosition = [-95.16305892510273, 754.3761479590994, 426.10491067642846]
+    renderView1.CameraFocalPoint = [43.11431452669897, 23.554652286151217, -46.78071794111179]
+    renderView1.CameraViewUp = [0.09790398117889891, -0.5275190119519493, 0.843882990999677]
+    renderView1.CameraParallelScale = 715.9367631158923
+    renderView1.Background = [0.0, 0.0, 0.0]
+    return renderView1
+
+def load_InputData():
+    '''
+    Read data from yaml input file
+    '''
+    Input_stream = open("InputVis_PV.yaml", 'r')
+    Input_data = yaml.load(Input_stream)
+    return Input_data
+
+def readData_Xdmf(xdmf_FilePath): 
+    '''
+    Read data from XDMF file specified in 
+    yaml file.
+    '''
+    try:
+        xDMFReader1 = XDMFReader(FileNames=xdmf_FilePath)
+    except FileNotFoundError:
+        sys.exit("No such file at file location specified")
+    xDMFReader1 = GetActiveSource()
+    renderView1 = GetActiveViewOrCreate('RenderView')
+    xDMFReader1_Display = GetDisplayProperties(xDMFReader1, view=renderView1)
+    return renderView1, xDMFReader1, xDMFReader1_Display
+
+def set_Representation(representation, renderView1, xDMFReader1_Display):
+    '''
+    Set representation and update view 
+    '''
+    xDMFReader1_Display.Representation = representation
+    renderView1.Update()
+    return renderView1, xDMFReader1_Display
+
+def set_colorMap(var, renderView1, xDMFReader1_Display, color_map):
+    '''
+    Set color map to one of preset ParaView color maps.
+    If color map entered is invalid, color map is set to 
+    default.
+    '''
+    ColorBy(xDMFReader1_Display, ('POINTS', var))
+    Var_LUT = GetColorTransferFunction(var)
+    try:
+        Var_LUT.ApplyPreset(color_map, True)
+    except:
+        print('Using default color map')
+        Var_LUT.ApplyPreset('Viridis (matplotlib)', True)
+    xDMFReader1_Display.SetScalarBarVisibility(renderView1, True)
+    xDMFReader1_Display.SetScaleArray = ['POINTS', var]
+    xDMFReader1_Display.ScaleTransferFunction = 'PiecewiseFunction'
+    return xDMFReader1_Display, renderView1, Var_LUT 
+
+def tetrahedralize(xDMFReader1, renderView1):
+    '''
+    Apply tetrahedralize filter and update view
+    '''
+    tetrahedralize1 = Tetrahedralize(Input=xDMFReader1)
+    Hide(xDMFReader1, renderView1)
+    xDMFReader1_Display = Show(tetrahedralize1, renderView1)
+    renderView1.Update()
+    return renderView1, xDMFReader1_Display
+
+def set_Opacity(var, fn_type, opacity_val, renderView1, xDMFReader1, Var_LUT):
+    '''
+    Set opacity based on option chosen in yaml input file:
```

Comment:
The function type can be either `Constant` or `Proportional`. (then expand on what that means)